File: csv-customer-file-generator/python-simple-approach/csv_customer_file_generator/writer.py
from pathlib import Path
from models import CustomerOrder


class Writer:
    @staticmethod
    def csv_file_writer(dir_path: Path, rows: list[CustomerOrder]):
        dir_path.mkdir(exist_ok=True)
        with open(dir_path / "customer-orders.csv", "w") as f:
            header = "CustomerId,ArticleId,OrderId,Timestamp"
            str_rows = "\n".join([",".join(list(map(str, row.__dict__.values()))) for row in rows])
            f.write(f"{header}\n{str_rows}")

File: csv-customer-file-generator/scala-simple-approach/src/main/scala/Writer.scala
import models.CustomerOrder

import java.io.{BufferedWriter, File, FileWriter}
import java.nio.file.{Files, Path}

object Writer {
  def csvFileWriter(path: Path, rows: List[CustomerOrder]): Unit = {
    if (!Files.exists(path)) Files.createDirectories(path)
    val file = new File(s"${path.toString}/customer-orders.csv")
    val writer = new BufferedWriter(new FileWriter(file))
    val lines = rows.map(_.toString).mkString("\n")
    writer.write("CustomerId,ArticleId,OrderId,Timestamp\n")
    writer.write(lines)
    writer.close()
  }
}
